packages/prio3/src/circuits/sum.ts

```diff
@@ -35,20 +35,35 @@ export class Sum extends Circuit<number> {
     );
   }
 
-  encode(measurement: number): bigint[] {
+  encode(measurement: number | bigint): bigint[] {
     if (
-      measurement !== Math.floor(measurement) ||
-      measurement < 0 ||
-      measurement >= 2 ** this.inputLen
+      typeof measurement === "number" &&
+      measurement !== Math.trunc(measurement)
     ) {
       throw new Error(
-        `measurement ${measurement} was not an integer in [0, ${
+        `measurement ${measurement} was not an integer [0, ${
```

Comment:
nit: the error in this branch is missing a word
```suggestion
        `measurement ${measurement} was not an integer in [0, ${
```

Reply:
Thanks!